src/searchbar.cpp

```diff
@@ -165,6 +170,44 @@ void SearchBarLineEdit::updateCompletion()
     fetchSuggestions(&SearchBarLineEdit::onInitialSuggestions);
 }
 
+void SearchBarLineEdit::fetchMoreSuggestions()
+{
+    fetchSuggestions(&SearchBarLineEdit::onAdditionalSuggestions);
+}
```

Comment:
This encapsulates a performance bug. `fetchSuggestions()` creates a temporary `SuggestionListWorker` which creates  `zim::SuggestionSearcher` that is discarded after a single use. In a more optimal implementation the `zim::SuggestionSearcher` should be reused for loading additional suggestions (since those incremental calls to `zim::SuggestionSearcher::getResults()` should be much cheaper compared to the first call). I am not implying that that optimization be implemented in this PR. I propose to do it myself after this PR is merged, but if you feel like becoming married to this suggestions feature and would like to take care of it to the end I will humbly stay away :)

Reply:
Let's do this separately.

Reply:
Definitely. But please add a TODO comment in the body of `SearchBarLineEdit::fetchMoreSuggestions()` in this PR.

Reply:
Please don't start working on this until the rest of your `kiwix-desktop` PRs are merged.